lb_film_and_user_recommender.py

```diff
@@ -0,0 +1,251 @@
+import urllib.request, urllib.parse, urllib.error
+import http
+import re
+import math
+
+def webpage_to_string(url):
+    """
+    Returns a string of the raw HTML for an entire web page.
+
+    :param url: (string or HTTPResponse object') The URL of the website as a string, or an HTTPResponse object (part of
+    the http library) that has been requested before being passed as an argument. This function handles either.
+    :return: (string) A string of the raw HTML for the URL provided.
+    """
+    if isinstance(url, str):
+        return str(list(urllib.request.urlopen(url)))
+    elif isinstance(url, http.client.HTTPResponse):
+        return str(list(url))
+    else:
+        return 'Error: URL provided is not a string or an HTTPResponse object'
```

Comment:
Very nice. Still don't need the `elif` and `else` there though, just `if` since the blocks return.